browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchClient.java

```diff
@@ -0,0 +1,135 @@
+        return new BrowserSwitchClient(config, activityFinder, persistentStore);
+    }
+
+    private BrowserSwitchClient(BrowserSwitchConfig config, ActivityFinder activityFinder, BrowserSwitchPersistentStore persistentStore) {
+        this.config = config;
+        this.activityFinder = activityFinder;
+        this.persistentStore = persistentStore;
+    }
+
+    void start(int requestCode, Intent intent, FragmentActivity activity, BrowserSwitchListener listener) {
+        Context applicationContext = activity.getApplicationContext();
+        String errorMessage = assertCanPerformBrowserSwitch(requestCode, applicationContext, intent);
+
+        if (errorMessage != null) {
+            BrowserSwitchResult result =
+                new BrowserSwitchResult(BrowserSwitchResult.STATUS_ERROR, errorMessage);
+            listener.onBrowserSwitchResult(requestCode, result,null);
+        } else {
+            BrowserSwitchRequest request =
+                    new BrowserSwitchRequest(requestCode, intent.getData(), BrowserSwitchRequest.PENDING);
+            persistentStore.putActiveRequest(request, activity);
+
+            applicationContext.startActivity(intent);
+        }
+    }
+
+    private String assertCanPerformBrowserSwitch(int requestCode, Context context, Intent intent) {
+        String errorMessage = null;
+
+        if (!isValidRequestCode(requestCode)) {
+            errorMessage = "Request code cannot be Integer.MIN_VALUE";
+        } else if (!isConfiguredForBrowserSwitch(context)) {
+            errorMessage =
+                "The return url scheme was not set up, incorrectly set up, " +
+                "or more than one Activity on this device defines the same url " +
+                "scheme in it's Android Manifest. See " +
+                "https://github.com/braintree/browser-switch-android for more " +
+                "information on setting up a return url scheme.";
+        } else if (!canOpenUrl(context, intent)) {
+            errorMessage = String.format("No installed activities can open this URL: %s", intent.getData().toString());
+        }
+
+        return errorMessage;
+    }
+
+    private boolean isValidRequestCode(int requestCode) {
+        return requestCode != Integer.MIN_VALUE;
+    }
+
+    private boolean isConfiguredForBrowserSwitch(Context context) {
+        Intent browserSwitchActivityIntent = config.createIntentForBrowserSwitchActivityQuery(context);
+        return activityFinder.canResolveActivityForIntent(context, browserSwitchActivityIntent);
+    }
+
+    private boolean canOpenUrl(Context context, Intent intent) {
+        return activityFinder.canResolveActivityForIntent(context, intent);
+    }
+
+    public BrowserSwitchRequest getActiveRequest(Context context) {
+        return persistentStore.getActiveRequest(context);
+    }
+
+    public BrowserSwitchRequest takeActiveRequest(Context context) {
```

Comment:
Why is this function needed? How is it different than `clearActiveRequest()`?

Also - the name `take` maybe is confusing